Add File Manager

Alliionart opened the pull request:
Integrates [alexusmai's](https://github.com/alexusmai/laravel-file-manager) file explorer into LK for staff to easily pick through the desired disks/folders. The file manager is accessible through:

- sitename.com/filemanager
- File manager on admin settings
- Tinymce Image Source Button

This updates Tinymce to v5.4.2 - so far I have not seen any conflicts with anything core in 3.0.0. But maybe need to watch out for any other extensions with ties to older versions. This should be compatible with future tinymce versions.

All methods require the user be staff AND have the "manage_files" power. The default accessible folder is the /files folder and nothing else. Though two public examples exist in the commented code in the config.

Security tested on all 3 with both staff without the power, staff with the power, normal members and guests. 
<img width="730" height="433" alt="fm-gallery-view" src="https://github.com/user-attachments/assets/0a2b0562-9eeb-4ca7-a08d-2109325a6e34" />
<img width="927" height="478" alt="file-manager-main" src="https://github.com/user-attachments/assets/b8e5d16e-2b9d-4f92-b796-c3202a123d3c" />

Alliionart committed:
stop this madness

Alliionart committed:
refactor: fix blade formatting

Alliionart committed:
Initial installs

Alliionart committed:
refactor: fix PHP styling

Alliionart committed:
refactor: fix blade formatting

Alliionart committed:
Update TinyMCE & Update FileManager Popup Callbacks

+ Updated: TinyMCE to v5.4.2
- Fix: Callback now functions properly when opening the source and double clicking an image. This will automatically add the URL to the source box and close the file manager.
- Updated: Now only staff should have access to the file manager in case there is secret content.

Alliionart committed:
refactor: fix blade formatting

Alliionart committed:
Add to LK Admin Screen

- Adds a button to the File Manager on the base LK file manager page under settings.

Alliionart committed:
Merge branch 'extension/file-explorer' of https://github.com/Alliionart/lorekeeper into extension/file-explorer

Alliionart committed:
Auth Update

- Updated: Logged in users will no longer get an error accessing a page with a WYSIWYG
+ Added: New power 'manage_files' that can be added to roles. DO NOT ADD TO NORMAL USERS (Unless you configure the explorer with items you are ok with them seeing). PLEASE.

Alliionart committed:
refactor: fix blade formatting

Alliionart committed:
Actually Fixed...

Now the File Manager will show if the user has the power to manage_files

Alliionart committed:
Merge branch 'extension/file-explorer' of https://github.com/Alliionart/lorekeeper into extension/file-explorer

Alliionart committed:
refactor: fix PHP styling

Alliionart committed:
Hide button

Alliionart committed:
Merge branch 'extension/file-explorer' of https://github.com/Alliionart/lorekeeper into extension/file-explorer

Alliionart committed:
refactor: fix blade formatting

Alliionart committed:
Security Fixes

Ensures double no-access to the file manager
- Removed: Public folder access

Alliionart committed:
refactor: fix PHP styling

itinerare commented on .gitignore:
```diff
@@ -9,6 +9,5 @@ Homestead.json
 Homestead.yaml
 npm-debug.log
 yarn-error.log
-*.env
```
This shouldn't be changed...

itinerare commented on routes/web.php:
```diff
@@ -11,6 +11,10 @@
 
+use App\Http\Controllers\FileManagerController;
+
+Route::get('filemanager', [FileManagerController::class, 'index']);
```
Until/unless the rest of the routes are updated to the new format, this should use the same format as the others.

itinerare commented on app/Http/Kernel.php:
```diff
@@ -64,6 +64,7 @@ class Kernel extends HttpKernel {
         'signed'        => \Illuminate\Routing\Middleware\ValidateSignature::class,
         'throttle'      => \Illuminate\Routing\Middleware\ThrottleRequests::class,
         'verified'      => \Illuminate\Auth\Middleware\EnsureEmailIsVerified::class,
+        'manage_files'  => \App\Http\Middleware\CheckPower::class.':manage_files',
```
Is there a reason this is set up like this rather than using the existing "power" middleware above?

itinerare commented on composer.json:
```diff
@@ -28,7 +29,8 @@
         "spatie/laravel-feed": "^4.1",
-        "spatie/laravel-honeypot": "^4.1"
+        "spatie/laravel-honeypot": "^4.1",
+        "tinymce/tinymce": "5.4.2"
```
It might be better to handle this via npm and include it in what mix handles... Probably out of scope for this PR, though.

itinerare replied:
All told, it might be easier to update tinyMCE *that* way in a separate PR, and *then* apply these changes on top... as it is, this is a tough PR to review because the signal/noise ratio is a bit dire due to the update changes.

Alliionart replied:
Would it be worth it to even update tinymce to a higher version like v7 for the [enhanced code editor](https://www.tiny.cloud/docs/tinymce/7/advcode/) to merge that into core?  Adding the [reference](https://www.tiny.cloud/docs/tinymce/latest/npm-projects/) here

itinerare replied:
I've had issues with tinyMCE versions past... 6.7.3, it looks like? where the toolbar likes to vanish. though it may be a quirk of vite or something. But, assuming that doesn't kick in, potentially?

preimpression replied:
@itinerare Did it load without the toolbar and take way too long to have the toolbar appear? I had that issue and seemed to resolve it by removing addons/plugins I didn't necessarily use (spoiler, emojis, etc) and now it loads much faster. Potentially a similar cause?

itinerare replied:
The toolbar just never loaded-- but this is, again, only on versions after the stated one, so not something that should impact LK unless you've gone out of your way to update it.

Alliionart committed:
PR Fixes

- Fix .gitignore issues
- Add file manager for home controller for routing
- remove unneeded middleware

Alliionart committed:
refactor: fix PHP styling

SpeedyD commented:
I'm having trouble understanding why this seems entirely necessary.. is there a way to disable this function entirely?
